components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -350,3 +354,39 @@ func assertIssuerIsReachable(host string) error {
 	}
 	return nil
 }
+
+func assertIssuerProvidesDiscovery(issuer string) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   2 * time.Second,
```

Comment:
Any reason to scope the timeout down to only 2 seconds? I think we can make this simpler for ourselves and use the context here to allow for as long as the client is happy to wait (or we can't reach) which can prevent errors in some higher latency envs

Reply:
Indeed. 
Done